apex/contrib/sparsity/sparse_masklib.py

```diff
@@ -22,33 +23,127 @@ def reshape_1d(matrix, m):
 
 """ return all possible m:n patterns in a 1d vector """
 valid_m4n2_1d_patterns = None
+valid_m16n8_1d_patterns = None
+valid_m32n8_1d_patterns = None
+valid_m32n16_1d_patterns = None
 def compute_valid_1d_patterns(m,n):
     # Early exit if patterns was already created.
     global valid_m4n2_1d_patterns
-
+    global valid_m16n8_1d_patterns
+    global valid_m32n8_1d_patterns
+    global valid_m32n16_1d_patterns
     if m==4  and n==2 and valid_m4n2_1d_patterns  is not None: return valid_m4n2_1d_patterns
-    patterns = torch.zeros(m)
-    patterns[:n] = 1
-    valid_patterns = torch.tensor(list(set(permutations(patterns.tolist()))))
-    if m == 4 and n == 2: valid_m4n2_1d_patterns = valid_patterns
+    if m==16  and n==8 and valid_m16n8_1d_patterns  is not None: return valid_m16n8_1d_patterns
+    if m==32  and n==8 and valid_m32n8_1d_patterns  is not None: return valid_m32n8_1d_patterns
+    if m==32  and n==16 and valid_m32n16_1d_patterns  is not None: return valid_m32n16_1d_patterns
+    valid_patterns = []
+    for i in list(combinations(range(0, m), n)):
+        cur_pattern = np.zeros(m, dtype=np.int32)
+        cur_pattern[list(i)] = 1
+        valid_patterns.append(cur_pattern)
+    valid_patterns = torch.Tensor(np.array(valid_patterns))
```

Comment:
Hi @LeiWang1999  - as I suggested in your previous [PR 1501](https://github.com/NVIDIA/apex/pull/1501), I'd prefer if the existing 2:4 mask candidate orders were maintained.  We can accomplish this with:
`valid_patterns = torch.Tensor(list(set([tuple(vp) for vp in valid_patterns])))`